### PR TITLE
Fgp 103/jwks and token update

### PR DESCRIPTION
**Pull request by iaincooper-tco**

### CODE REVIEW

File: src/main/resources/db/migration/V1_3__add_token_blacklist_sequence.sql [hunk 1]
```diff
@@ -0,0 +1,3 @@
+CREATE SEQUENCE token_blacklist_seq
```
Comment (GavCookCO) [inline src/main/resources/db/migration/V1_3__add_token_blacklist_sequence.sql:1]: This has already been done as part of a migration to fix the previous migration that I broke so suspect this won't work when merged into develop

File: src/main/java/gov/cabinetofice/gapuserservice/web/LoginController.java [hunk 2]
```diff
@@ -65,18 +74,26 @@ public RedirectView redirectAfterColaLogin(final @CookieValue(name = REDIRECT_UR
...
+        }
         final Cookie userTokenCookie = WebUtil.buildCookie(
-                new Cookie(USER_SERVICE_COOKIE_NAME, customJwtService.generateToken()),
+                new Cookie(USER_SERVICE_COOKIE_NAME, customJwtService.generateToken(claims)),
                 Boolean.TRUE,
                 Boolean.TRUE
         );
 
         response.addCookie(userTokenCookie);
+        System.out.println(userTokenCookie.getValue());
```
Comment (dominicwest) [inline src/main/java/gov/cabinetofice/gapuserservice/web/LoginController.java:91]:
```suggestion

```

File: src/main/java/gov/cabinetofice/gapuserservice/web/LoginController.java [hunk 3]
```diff
@@ -96,11 +113,20 @@ public RedirectView logout(
...
+                                   final @RequestParam String redirectUrl) {
         jwtBlacklistService.addJwtToBlacklist(currentToken);
 
-        final String newToken = customJwtService.generateToken();
+        final DecodedJWT decodedJWT = JWT.decode(currentToken);
+
+        Map<String, String> claims = new HashMap<>();
+        for (Map.Entry<String, Claim> entry : decodedJWT.getClaims().entrySet()) {
+            claims.put(entry.getKey(), entry.getValue().asString());
+            System.out.println(entry.getKey() + ":" + entry.getValue());
```
Comment (dominicwest) [inline src/main/java/gov/cabinetofice/gapuserservice/web/LoginController.java:126]:
```suggestion
```

Comment (iaincooper-tco) [reply src/main/java/gov/cabinetofice/gapuserservice/web/LoginController.java:125]: also nice catch

**Review by dominicwest — APPROVED** [review]

**Pull request merged**